### ISSUE TITLE
neovim does not start on windows server 2022

### ISSUE THREAD

**Issue by fekir** [neovim/neovim#29516]
### Problem

I've installed neovim on Windows server 2022, but the program does not seem to start.

### Steps to reproduce

 * Install neovim
 * Execute (from powershell) `& "C:\Program Files\Neovim\bin\nvim.exe"`
 * Execute (from powershell) `& "C:\Program Files\Neovim\bin\nvim.exe" --clean`
 * Execute (from powershell) `& "C:\Program Files\Neovim\bin\nvim.exe" --help`
 * Execute (from powershell) `& "C:\Program Files\Neovim\bin\nvim.exe" --clean --help`

### Expected behavior

I would expect neovim to start.
When invoked with `--help` to show the help message

### Neovim version (nvim -v)

0.10.0 (nvim -v does not print anything)

### Vim (not Nvim) behaves the same?

vim (VIM - Vi IMproved 9.1) starts correctly and is able to edit files

### Operating system/version

Windows Server 2022 (update june 2024)

### Terminal name/version

powershell

### $TERM environment variable

""

### Installation

setup

**Comment by fekir** [comment]
Forgot to add, windows server 2022 is installed without "Desktop Experience".

It means, for example, that "explorer.exe" is not available.

**Comment by fekir** [comment]
I've compared the output of dependency walker on Windows Server with and without the Desktop component.

On the system without desktop component, it seems that vcruntime140.dll is not found.


![image](https://github.com/neovim/neovim/assets/11912465/b2480adb-bfa1-419d-8781-599edcb9e717)


Right side is Windows Server 2022 without Desktop experience.

Shouldn't vcruntime140.dll get installed by the installer?

**Comment by justinmk** [comment]
> Shouldn't vcruntime140.dll get installed by the installer?

No. That is generally a one-time global install that you must perform yourself.

btw, what does `winver` report as your exact Windows version? From `:help supported-platforms` :

> Note: Windows 10 "Version 1809" or later is required for |:terminal|. To check your Windows version, run the "winver" command and look for "Version xxxx" (NOT "OS Build").

**Comment by fekir** [comment]
@justinmk 

winver does not seem to be available on windows server 2022

> winver : The term 'winver' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.


Also `:help supported-platforms` does not help much if nvim does not start.

While updating the documentation is better than the status quo, it would be more user-friendly if 

 * neovim does not depend on `vcruntime140.dll` (static build? should be achievable with `/MT` flag)
 * neovim installer installs `vcruntime140.dll` (in the app folder, not globally)
 * neovim installer warns if `vcruntime140.dll` is missing

The first two options would be preferred in order to having nvim just working after installation.

Otherwise the experience is pretty bad, as the executable seems to do nothing, while in reality the application does not start without any error message

**Comment by justinmk** [comment]
> Also `:help supported-platforms` does not help much if nvim does not start.

But Windows Server is missing all kinds of things (winver, explorer, vcrt, ...). Though I know it's common on remote cloud systems, and it would be good to have a better experience there.

> neovim does not depend on vcruntime140.dll (static build? should be achievable with /MT flag)

Perhaps we should consider that. But it makes the package much bigger for all users, which is unfortunate since the common case is that Windows has its standard tools available.

I'm not sure it makes sense for a "portable" build to target a minimal system that is missing its standard tools. Instead, a package manager (scoop, winget) should be able to pull in what's needed for such a system.

> neovim installer warns if vcruntime140.dll is missing

That's definitely where we should start. Is there any reference implementation for this? Does https://wixtoolset.org/ have a way to do it?

**Comment by fekir** [comment]
> Though I know it's common on remote cloud systems, and it would be good to have a better experience there.

And especially on remote system, and system without UI I want to use a command-line editor ;)

> But it makes the package much bigger for all users, which is unfortunate since the common case is that Windows has its standard tools available.

Then maybe conditionally installing the missing dll in the app folder makes more sense

> Instead, a package manager (scoop, winget) should be able to pull in what's needed for such a system.

Unfortunately, windows is lacking one

**Comment by justinmk** [comment]
> Unfortunately, windows is lacking one

[scoop](https://scoop.sh/) and [winget](https://github.com/microsoft/winget-cli) are windows packages managers. But idk if they install vcruntime140.dll when needed.

**Comment by bfredl** [comment]
> Installation: setup

This does not say anything. Is this official binaries or from a package manager?

Our own binaries should only depend on UCRT, not any version of vcruntime.

**Comment by fekir** [comment]
@bfredl 

> > Installation: setup

I'm not sure who you are replying to here.

> This does not say anything. Is this official binaries or from a package manager?

With "For me, the official setup is the msi file/zip archive." I meant the setup and archive that can be downloaded from github, not what winget or scoop do on some platforms.
I did only test the msi installer from here.

> Our own binaries should only depend on UCRT, not any version of vcruntime.

nvim depends on vcruntime140.dll, as shown in my third comment, I also tested the current release yesterday.
Maybe not nvim directly, but one of its dependencies.

Installing  https://aka.ms/vs/17/release/vc_redist.x64.exe fixes the issue; nvim starts.

**Comment by justinmk** [comment]
> nvim depends on vcruntime140.dll

did you try the latest nightly? https://github.com/neovim/neovim/releases/tag/nightly 
there have been changes to the build compared to the stable build.

**Comment by fekir** [comment]
> > nvim depends on vcruntime140.dll
> 
> did you try the latest nightly? https://github.com/neovim/neovim/releases/tag/nightly there have been changes to the build compared to the stable build.

No, I can try

> 
> from https://learn.microsoft.com/en-us/cpp/windows/universal-crt-deployment?view=msvc-170
> 
> > The Universal CRT is a Recommended update for all supported Microsoft Windows operating systems, so by default, most machines install it as part of the regular update process.

I read that; the article is of 2022, and thus it might be out of date.
What dll is the Universal CRT?

Also Windows Server 2022 without Desktop experience is a stripped down windows version, thus not all articles on learn.microsoft.com hold.

**Comment by fekir** [comment]
@justinmk 

> > nvim depends on vcruntime140.dll
> 
> did you try the latest nightly? https://github.com/neovim/neovim/releases/tag/nightly there have been changes to the build compared to the stable build.


Tested NVIM v0.11.0-dev-1450+g59cbe64023 (f5859cd32d31944e265c8abeedc75c1e01e77bfa9f0c644be8953b2b572fad62  nvim-win64.zip).

nvim depends directly on vcruntime140.dll, and lua51.dll depends on vcruntime140.dll too.

**Comment by adrianstaniec** [comment]
`scoop install vcredist`
fixed it for me
